### PR TITLE
feat: add method to retrieve bindingObserver subscriptions

### PR DESCRIPTION
**Pull request by nicholasrice**
# Pull Request

## 📖 Description
This PR adds a method to `BindingObserver` that allows inspection of the internal `SubscriptionRecord`'s used being used by a `BindingObserver`. This allows us to actually inspect which objects & values are being observed by a `BindingObserver`.

### General

- [ ] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: packages/web-components/fast-element/src/observation/observable.ts [hunk 1]
```diff
@@ -377,6 +377,12 @@ export interface BindingObserver<TSource = any, TReturn = any, TParent = any>
      * Unsubscribe from all dependent observables of the binding.
      */
     disconnect(): void;
+
+    /**
+     * Gets {@link SubscriptionRecord|SubscriptionRecords} that the {@link BindingObserver}
+     * is observing.
+     */
+    subscriptionRecords(): IterableIterator<SubscriptionRecord>;
```
Comment (nicholasrice) [inline packages/web-components/fast-element/src/observation/observable.ts:385]: @EisenbergEffect I wasn't exactly sure what to name this so if you have something better in mind let me know. I also thought about simply `BindingObserver.records()`

Comment (EisenbergEffect) [reply packages/web-components/fast-element/src/observation/observable.ts:385]: Let's go with `records` if that's ok with you.

Comment (EisenbergEffect) [reply packages/web-components/fast-element/src/observation/observable.ts:385]: Also, I think we may need to export `SubscriptionRecord` or some version of it. Do we want to export that directly or have a version that we export with only the `propertySource` and `propertyName` properties? Should we include `notifier` as well?

**Review by EisenbergEffect — CHANGES_REQUESTED** [review]
Looks good. Just had one point I wanted us to ponder before merging.

File: packages/web-components/fast-element/src/observation/observable.ts (hunk 1 above)
Comment (nicholasrice) [reply packages/web-components/fast-element/src/observation/observable.ts:385]: The the `notifier` can get obtained from `Observable.getNotifier(propertySource)` right? If so I think we can probably just expose `propertySource` and `propertyName`

Comment (nicholasrice) [reply packages/web-components/fast-element/src/observation/observable.ts]: @EisenbergEffect changed method name to `records()` and exported an `ObservationRecord` interface

**Review by EisenbergEffect — APPROVED** [review]
Looks good. If you can, add some docs on the new interface.

**Comment by nicholasrice** [comment]
> Looks good. If you can, add some docs on the new interface.

Added some interface docs and updated observable docs for doc site. Lemme know if that all sounds right or not.

File: packages/web-components/fast-element/docs/guide/observables-and-state.md [hunk 2]
```diff
@@ -191,4 +191,20 @@ const handler = {
...
+To inspect which observable objects and properties were accessed from a `BindingObserver`, you can get the observation records from `BindingObserver.records()` after observing the binding.
+
+**Example: Getting observation records**
+```ts
+const binding = (x: MyClass) => x.someBoolean ? x.valueA : x.valueB;
+const bindingObserver = Observable.binding(binding);
+const value = bindingObserver.observe({}, defaultExecutionContext);
+
+for (const record of bindingObserver.records()) {
+  // Do something with the bindings observable dependencies
```
Comment (nicholasrice) [inline packages/web-components/fast-element/docs/guide/observables-and-state.md:207]:
```suggestion
  // Do something with the binding's observable dependencies
```

**Review by janechu — APPROVED** [review]

**Comment by EisenbergEffect** [comment]
Thanks @nicholasrice !

**Pull request merged**